examples/decoding/plot_receptive_field.py

```diff
@@ -0,0 +1,120 @@
+"""
+=============================================
+Receptive field estimation on continuous data
+=============================================
+
+This demonstrates how an encoding model can be fit with multiple continuous
+inputs. In this case, we simulate the model behind a spectro-temporal receptive
+field (or STRF). First, we create a linear filter that maps patterns in
+spectro-temporal space onto an output, representing neural activity. We fit
+a receptive field model that attempts to recovre the original linear filter
+that was used to create this data.
+
+Estimation of spectro-temporal and spatio-temporal receptive fields using
+modeling with continuous inputs is described in:
+
+    Theunissen, F. E. et al. Estimating spatio-temporal receptive
+    fields of auditory and visual neurons from their responses to
+    natural stimuli. Network 12, 289-316 (2001).
+
+    Willmore, B. & Smyth, D. Methods for first-order kernel
+    estimation: simple-cell receptive fields from responses to
+    natural scenes. Network 14, 553-77 (2003).
+"""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#
+# License: BSD (3-clause)
+
+import mne
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy import stats as stt
+from mne.decoding.receptive_field import ReceptiveField, delay_time_series
+from sklearn.preprocessing import scale
+np.random.seed(1337)
+
+# Path to audio stimuli
+audio_path = 'audio-epo.fif'
+
+# --- Create a receptive field ---
+# Define the time/freq space for our receptive field
+n_freqs = 32
+n_lags = 20
+lags = np.linspace(0, .4, n_lags)
+freqs = np.logspace(2, np.log10(6000), n_freqs)
+grid = np.array(np.meshgrid(lags, freqs))
+grid = grid.swapaxes(0, -1).swapaxes(0, 1)
+
+# Receptive field parameters
+means_hi = [.1, 2000]
+means_lo = [.15, 3000]
+cov = [[.0005, 0], [0, 200000]]
+gauss_hi = stt.multivariate_normal.pdf(grid, means_hi, cov)
+gauss_lo = -1 * stt.multivariate_normal.pdf(grid, means_lo, cov)
+weights = gauss_hi + gauss_lo  # Combine to create the "true" STRF
+
+# --- Use the receptive field to simulate a response to sound ---
+# Load the auditory input and turn it into a spectrogram
+# XXX only pulling a few trials to speed up
+audio = mne.read_epochs(audio_path)
+n_ep = len(audio)
+
+n_decim = 8
+spec = mne.time_frequency.tfr_morlet(audio, freqs, 10, return_itc=False,
+                                     average=False, picks=[0], decim=n_decim)
+sfreq_new = audio.info['sfreq'] / n_decim
+X = spec.data[:6, 0, ...]
+n_times = X.shape[-1]
+
+# Delay the spectrogram according to lags so it can be combined w/ the STRF
+X_del = delay_time_series(X, lags, sfreq_new)
+X_del = X_del.swapaxes(0, 1).reshape([n_ep, -1, n_times])
```

Comment:
Add a comment here saying that ReceptiveField requires data to be shaped as `n_epochs, n_features, n_times`, thus why axes are swapped. I do wonder if there is a cleaner way to do this, either by making `delay_time_series` smarter or `ReceptiveField`. People might have a hard time with the reshaping and axis swapping.